db/migrate/20200527101013_add_unique_indexes_to_articles_slug_feed_source_url_canonical_url_body_markdown.rb

```diff
@@ -0,0 +1,40 @@
+class AddUniqueIndexesToArticlesSlugFeedSourceUrlCanonicalUrlBodyMarkdown < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def up
+    if index_exists?(:articles, :feed_source_url)
+      remove_index :articles, column: :feed_source_url, algorithm: :concurrently
+    end
+
+    # needed for the digest() function, checks if the extension already exists
+    ActiveRecord::Base.connection.enable_extension("pgcrypto")
```

Comment:
Should we add an `unless` statement here like they do in the edge docs? https://edgeguides.rubyonrails.org/active_record_postgresql.html#uuid-primary-keys
```ruby 
enable_extension 'pgcrypto' unless extension_enabled?('pgcrypto')
```

Looks like we can remove the `ActiveRecord::Base.connection` as well.

Reply:
as the comment says, that check is builtin in Rails :-D 

> is there any chance that this extension can't be enabled for some reason on a Postgres db

nope, it's builtin - https://www.postgresql.org/docs/11/pgcrypto.html.

If we ever encounter a custom version of PostgreSQL that has that module disabled somehow I think it should explode as pgcrypto contains all the cryptographic functions anyway